mysql-test/suite/galera/t/galera_restart_replica.test

```diff
@@ -0,0 +1,138 @@
+# while node_4 is dead
+SET GLOBAL wsrep_provider_options = 'pc.ignore_sb=true';
+SET GLOBAL wsrep_provider_options = 'pc.weight=2';
+
+--echo # Waiting table and rows to replicate to node_3
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 't1'
+--source include/wait_condition.inc
+--let $wait_condition = SELECT COUNT(*) = 50000 FROM t1
+--source include/wait_condition.inc
+SELECT COUNT(*) FROM t1;
+
+--connection node_4
+--source include/sync_with_master_gtid.inc
+--echo # Waiting table and rows to replicate to node_4
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 't1'
+--source include/wait_condition.inc
+--let $wait_condition = SELECT COUNT(*) = 50000 FROM t1
+--source include/wait_condition.inc
+SELECT COUNT(*) FROM t1;
+
+--connection node_4
+# Isolate node_4 from the group and wait until wsrep_ready becomes OFF.
+SET GLOBAL wsrep_provider_options = 'gmcast.isolate = 1';
+SET SESSION wsrep_sync_wait = 0;
+--let $wait_condition = SELECT VARIABLE_VALUE = 'OFF' FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_ready'
+--source include/wait_condition.inc
+
+--echo # Verify that graceful shutdown succeeds.
+--source include/shutdown_mysqld.inc
+
+--connection node_3
+--echo # Waiting until node_4 is not part of cluster anymore
+--let $wait_condition = SELECT VARIABLE_VALUE = 1 FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+--source include/wait_condition.inc
+--let $wait_condition = SELECT VARIABLE_VALUE = 'Primary' FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_status';
+--source include/wait_condition.inc
+
+#
+# Add some writes to cluster1
+#
+--connection node_1
+--echo # Add writes to cluster
+--disable_query_log
+INSERT INTO t1 SELECT seq, 'test2' from seq_70000_to_80000;
+--enable_query_log
+SELECT COUNT(*) FROM t1;
+--source include/save_master_gtid.inc
+
+--connection node_4
+--echo # Start node_4 again
+--source include/start_mysqld.inc
+--source include/sync_with_master_gtid.inc
+
+SET GLOBAL wsrep_provider_options = 'gmcast.isolate=0';
+--let $wait_condition = SELECT VARIABLE_VALUE = 'Primary' FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_status';
+--source include/wait_condition.inc
+--let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+--source include/wait_condition.inc
+--let $wait_condition = SELECT VARIABLE_VALUE = 'ON' FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_ready';
+--source include/wait_condition.inc
+--let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+--source include/wait_condition.inc
+
+--connection node_4
+SELECT COUNT(*) FROM t1;
+
+--connection node_3
+SET GLOBAL wsrep_provider_options = 'pc.weight=1';
+SELECT COUNT(*) FROM t1;
+
```

Comment:
this would be non-deterministic as well, so wait_condition to expect the exact number of rows to be replicated is suggested here

Reply:
yes, will fix